crutest/src/main.rs

```diff
@@ -1924,77 +1990,118 @@ async fn replace_workload(
     guest: &Arc<Guest>,
     wtq: &mut WhenToQuit,
     ri: &mut RegionInfo,
-    full_targets: &[SocketAddr],
-    work: usize,
+    targets: Vec<SocketAddr>,
     fill: bool,
 ) -> Result<()> {
-    assert!(full_targets.len() == 4);
-
-    let mut preload_wtq = WhenToQuit::Count { count: 100 };
-    let mut replace_wtq = WhenToQuit::Count { count: work };
-
-    let mut c = 1;
-    let mut old_ds = 0;
-    let mut new_ds = 3;
-    loop {
-        println!("[{c}] Replace loop starts");
-        if fill {
-            fill_sparse_workload(guest, ri).await?;
-        }
-        generic_workload(guest, &mut preload_wtq, ri, false).await?;
+    assert!(targets.len() == 4);
 
-        println!(
-            "[{c}] Replacing DS {old_ds}:{} with {new_ds}:{}",
-            full_targets[old_ds], full_targets[new_ds],
-        );
-        let res = guest
-            .replace_downstairs(
-                Uuid::new_v4(),
-                full_targets[old_ds],
-                full_targets[new_ds],
-            )
-            .await;
-
-        match res {
-            Ok(ReplaceResult::Started) => {}
-            x => {
-                bail!("[{c}] Failed replace: {:?}", x);
-            }
-        }
-        old_ds = (old_ds + 1) % 4;
-        new_ds = (new_ds + 1) % 4;
-
-        generic_workload(guest, &mut replace_wtq, ri, false).await?;
+    if fill {
+        fill_sparse_workload(guest, ri).await?;
+    }
+    // Make a copy of the stop at counter if one was provided so the
+    // IO task and the replace task don't have to share wtq
+    let stop_at = match wtq {
+        WhenToQuit::Count { count } => Some(*count),
+        _ => None,
+    };
 
-        // Wait for all IO to settle down before we continue
+    // Start up a task to do the replacement workload
+    let (work_stop_tx, mut work_stop_rx) = mpsc::channel(1);
+    let (work_count_tx, mut work_count_rx) = mpsc::channel(1);
+    let guest_c = guest.clone();
+    let handle = tokio::spawn(async move {
+        let mut c: usize = 1;
```

Comment:
Nit: `c` is only used as a counter variable, so the `loop {` could instead be `for c in 1.. {`